### PR TITLE
Change modal position to a fixed vertical distance from the top of the window

### PR DESCRIPTION
**Pull request by abey79**
### What

This avoids the modal moving vertically when content size changes.

**NOTE**: this PR would trigger the bug fixed in https://github.com/emilk/egui/pull/3721, so it needs an egui release before being released. This is avoided by salting the modals' ID such as to in invalidate existing caches.

https://github.com/rerun-io/rerun/assets/49431240/d2d472b5-7af2-4e95-9da1-a671fc80629f

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4700/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4700/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4700/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4700)
- [Docs preview](https://rerun.io/preview/e159b005a7a1d5f44e608013f00c1d479b7c341c/docs) 
- [Examples preview](https://rerun.io/preview/e159b005a7a1d5f44e608013f00c1d479b7c341c/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by emilk** [comment]
> this PR is very likely to trigger the bug fixed in

you should be able to circumvent that by giving the area new id than whats already persisted

File: crates/re_ui/src/modal.rs [hunk 1]
```diff
@@ -109,10 +130,16 @@ impl Modal {
 
         let mut open = ui.input(|i| !i.key_pressed(egui::Key::Escape));
 
+        let screen_height = ui.ctx().screen_rect().height();
+        let modal_vertical_margins = (75.0).at_most(screen_height * 0.1);
+
         let mut window = egui::Window::new(&self.title)
```
Comment (emilk) [inline crates/re_ui/src/modal.rs:136]:
```suggestion
        let mut window = egui::Window::new(&self.title)
            .id(egui::Id::new("modal", &self.title)) // Workaround for https://github.com/emilk/egui/pull/3721 until we make a new egui release
```

Comment (abey79) [reply crates/re_ui/src/modal.rs:136]: Works (well—to the extent that it invalidates any existing cache).

**Review by emilk — APPROVED** [review]

File: crates/re_ui/src/modal.rs (hunk 1 above)
Comment (abey79) [reply crates/re_ui/src/modal.rs:136]: Yes, that's correct. So we should be good.

**Pull request merged**